lib/Parser.js

```diff
@@ -1396,6 +1388,214 @@ class Parser extends Tapable {
 		}
 	}
 
+	_prewalkNode(node) {
+		switch(node.type) {
+			case "BlockStatement":
+				this.prewalkBlockStatement(node);
+				break;
+			case "ClassDeclaration":
+				this.prewalkClassDeclaration(node);
+				break;
+			case "DoWhileStatement":
+				this.prewalkDoWhileStatement(node);
+				break;
+			case "ExportAllDeclaration":
+				this.prewalkExportAllDeclaration(node);
+				break;
+			case "ExportDefaultDeclaration":
+				this.prewalkExportDefaultDeclaration(node);
+				break;
+			case "ExportNamedDeclaration":
+				this.prewalkExportNamedDeclaration(node);
+				break;
+			case "ForInStatement":
+				this.prewalkForInStatement(node);
+				break;
+			case "ForOfStatement":
+				this.prewalkForOfStatement(node);
+				break;
+			case "ForStatement":
+				this.prewalkForStatement(node);
+				break;
+			case "FunctionDeclaration":
+				this.prewalkFunctionDeclaration(node);
+				break;
+			case "IfStatement":
+				this.prewalkIfStatement(node);
+				break;
+			case "ImportDeclaration":
+				this.prewalkImportDeclaration(node);
+				break;
+			case "LabeledStatement":
+				this.prewalkLabeledStatement(node);
+				break;
+			case "SwitchStatement":
+				this.prewalkSwitchStatement(node);
+				break;
+			case "TryStatement":
+				this.prewalkTryStatement(node);
+				break;
+			case "VariableDeclaration":
+				this.prewalkVariableDeclaration(node);
+				break;
+			case "WhileStatement":
+				this.prewalkWhileStatement(node);
+				break;
+			case "WithStatement":
+				this.prewalkWithStatement(node);
+				break;
+		}
+	}
+
+	_walkNode(node) {
+		switch(node.type) {
+			case "ArrayExpression":
+				this.walkArrayExpression(node);
+				break;
+			case "ArrayPattern":
+				this.walkArrayPattern(node);
+				break;
+			case "ArrowFunctionExpression":
+				this.walkArrowFunctionExpression(node);
+				break;
+			case "AssignmentExpression":
+				this.walkAssignmentExpression(node);
+				break;
+			case "AssignmentPattern":
+				this.walkAssignmentPattern(node);
+				break;
+			case "AwaitExpression":
+				this.walkAwaitExpression(node);
+				break;
+			case "BinaryExpression":
+				this.walkBinaryExpression(node);
+				break;
+			case "BlockStatement":
+				this.walkBlockStatement(node);
+				break;
+			case "CallExpression":
+				this.walkCallExpression(node);
+				break;
+			case "CatchClause":
```

Comment:
not covered by tests.

This is not entered via walkStatement, but via walkTryStatement directly. Can be removed?

Reply:
Yes sure